src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBuilder.java

```diff
@@ -220,7 +223,9 @@ public void reduceNeededResource(Block res, int amount)
         }
         else
         {
-            this.neededResources.put(res, preAmount - amount);
+            IBlockState state = res.getStateFromMeta(metaData);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "getStateFromMeta"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'state' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)